src/librustc_resolve/diagnostics.rs

```diff
@@ -425,6 +521,55 @@ match 0 {
 ```
 "##,
 
+E0419: r##"
+An unknown enum variant, struct or const was used. Example of
+erroneous code:
+
+```
+match 0 {
+    Something::Foo => {} // error: unresolved enum variant, struct
+                         //        or const `Foo`
+}
+```
+
+Please verify you didn't misspell it or the enum variant, struct
+or const was well declared and/or imported. Example:
+
+```
+enum Something {
+    Foo,
+    NotFoo,
+}
+
+match Something::NotFoo {
+    Something::Foo => {} // ok!
+    _ => {}
+}
+```
+"##,
+
+E0423: r##"
+A `struct` variant name was used like a function name. Example of
```

Comment:
Yeah, I think the error message should be reverted to what it is in stable and beta as well.

I’m fine with merging this as it is currently and changeing the explanation in the PR which updates the error message.